lib/nethealth/nethealth.go

```diff
@@ -484,7 +482,7 @@ func (s *Server) processAck(e messageWrapper) error {
 		return nil
 	default:
 		//unexpected / unknown
-		return trace.BadParameter("received unexpected icmp message type")
+		return trace.BadParameter("received unexpected icmp message type").(trace.Error).AddField("type", e.message.Type)
```

Comment:
I've [updated](https://github.com/gravitational/trace/pull/50) trace type constructors to return `Error` so you don't need to type-cast:
```go
return trace.BadParameter("received unexpected icmp message type").AddField("type", e.message.Type)
```